src/Application/SearchUrlBuilder.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Application;
+
+use MediaWiki\Utils\UrlUtils;
+
+class SearchUrlBuilder {
+
+	/** @var array<string, string> */
+	private array $urlParts = [];
+
+	/** @var array<string, string> */
+	private array $urlQuery = [];
+
+	public function __construct(
+		private readonly UrlUtils $urlUtils
+	) {
+	}
+
+	public function setUrlParts( string $url ): void {
+		$this->urlParts = $this->urlUtils->parse( $url ) ?? [];
+	}
+
+	/**
+	 * @return array<string, string>
+	 */
+	public function getUrlParts(): array {
+		return $this->urlParts;
+	}
+
+	public function setUrlQuery(): void {
+		$this->urlQuery = wfCgiToArray( (string)$this->urlParts['query'] );
+	}
```

Comment:
Since this is a setter, I was expecting a parameter.
It also looks like this method is always called after `setUrlParts`, so perhaps there needs to be a single public method like `setUrl( string $url )` which implements or calls both.